packages/snaps-sdk/src/internals/structs.ts

```diff
@@ -99,16 +98,27 @@ export function typedUnion<Head extends AnyStruct, Tail extends AnyStruct[]>(
     )
     .flat(Infinity);
   const types = flatStructs.map(({ schema }) => schema.type.type);
+  const structSet = flatStructs.reduce<Record<string, Struct>>(
```

Comment:
If performance is important, using `reduce` seems to be not optimal. We can likely combine it with the loop above as well, using a plain for-loop.

Reply:
Hmm, potentially yes. Performance is probably less important outside of the struct implementation though as the outside logic shouldn't run too many additional times (with the exception of when used with `selectiveUnion`)

Reply:
Maybe not worth it for now?

Reply:
Should be easy enough to change, but I'll leave it up to you.

```ts
const types = [];
const structs = new Map();

for (const struct of flatStructs) {
  const type = struct.schema.type.type;

  types.push(type);
  structs.set(JSON.parse(type), struct);
}
```

Reply:
Yeah, I ran it quickly and it doesn't seem to make much difference, not worth it right now IMO